### PR TITLE
fix: all day event issues

### PR DESCRIPTION
**Pull request by atymic**
## Context and Purposes

See https://github.com/spatie/calendar-links/issues/183

This PR contains two main fixes. Tests included. 
cc @freekmurze @alies-dev 

## Ensuring all all day events are only handled in the UTC timezone
Most of the bugs in the above ticket are caused by the fact that the generators convert the passed start date to UTC, then format. This means that `2024-01-25 00:00:00+11:00` becomes `2024-01-24` when formatted, leading to the wrong date.

**Fix**: When `allday = true` dates are only handled in UTC

## `create` method `allDay` flag behaviors update
Not sure what ya'll think about this one, but from talking to users, when they select date `2024-01-25` -  `2024-01-30`, they are expecting it to be inclusive. The issue is the library (assuming other bug above is ignored) handles it as exclusive.

When passing `startdate=2024-01-25, days=5` to `createAllDay`, event produced is  `2024-01-25` -  `2024-01-29` which makes sense (5 full days, as last day is inclusive)

When passing `startdate=2024-01-25, enddate=2024-01-30` to `create`, currently event produced is  `2024-01-25` -  `2024-01-29` which does not make sense. This PR would change that to produce a 6 day event, `2024-01-25` -  `2024-01-30`.

I guess this is a breaking change, so happy to hide this one behind a flag on the create method to avoid that.

### CODE REVIEW

**Comment by atymic** [comment]
cc @koposzek @parisiam

**Review by alies-dev — APPROVED** [review]
what's a great fix and amazing PR description! My complements, @atymic ! 💪

File: src/Link.php [hunk 1]
```diff
@@ -74,6 +82,11 @@ public static function create(string $title, \DateTimeInterface $from, \DateTime
      */
     public static function createAllDay(string $title, \DateTimeInterface $fromDate, int $numberOfDays = 1): self
     {
+        // In cases where the from date is not UTC, make sure it's UTC, size all day events are floating and non UTC dates cause bugs in the generators
+        if ($fromDate->getTimezone() !== new \DateTimeZone('UTC')) {
```
Comment (vever001) [inline src/Link.php:86]: I'm also having issues with the "All day" and gmail.
But does this line make sense? See https://3v4l.org/FAnI5
This is I believe 1 issue but I'm still having wrong dates with gmail besides this.

Comment (vever001) [reply src/Link.php:86]: Wouldn't this make more sense?
https://gist.github.com/vever001/98cba132c56c8b864126390164899bc3

Comment (vever001) [reply src/Link.php:86]: See https://github.com/spatie/calendar-links/pull/192

**Pull request merged**